### PR TITLE
Added leetcode solutions 18-20

### PR DESCRIPTION
**Pull request by Hemav009**
## Related Issue

Fix: #630 

## Description

Added leet code solutions 18-20

## Type of PR

- [ ] Bug fix
- [x] Feature enhancement
- [ ] Documentation update
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)
image

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

**Comment by Hemav009** [comment]
@Ajay-Dhangar  hey can you please explain what this error is?

**Comment by Ajay-Dhangar** [comment]
> @Ajay-Dhangar hey can you please explain what this error is?

sure, I will help you for debug to code

File: dsa-solutions/lc-solutions/0000-0099/0018- FourSum.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
+---
+id: FourSum
+title: 4Sum (LeetCode)
+sidebar_label: 0018-FourSum
+tags:
+  - Two Pointers
+description: Given an array nums of n integers, return an array of all the unique quadruplets [nums[a], nums[b], nums[c], nums[d]]
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0000-0099/0018- FourSum.md:7]: ### Bad practice

```md
description: Given an array nums of n integers, return an array of all the unique quadruplets [nums[a], nums[b], nums[c], nums[d]]
```

### Good practice

```md
description: "Given an array nums of n integers, return an array of all the unique quadruplets [nums[a], nums[b], nums[c], nums[d]]"
```

File: dsa-solutions/lc-solutions/0000-0099/0018- FourSum.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+#### Example 1
+
+- **Input:** `nums = [1,0,-1,0,-2,2], target = 0`
+- **Output:** `[[-2,-1,1,2],[-2,0,0,2],[-1,0,0,1]]`
+
+#### Example 2
+
+- **Input:** `nums = [2,2,2,2,2], target = 8`
+- **Output:** `[[2,2,2,2]]`
+
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0000-0099/0018- FourSum.md:37]: ### Basic Practice

```md
### Constraints

- `1 <= nums.length <= 200`
- `-10^9 <= nums[i] <= 10^9`
- `-10^9 <= target <= 10^9`
```

### Good Practice

```md
### Constraints

- $1 <= nums.length <= 200$
- $-10^9 <= nums[i] <= 10^9$
- $-10^9 <= target <= 10^9$
```

File: dsa-solutions/lc-solutions/0000-0099/0019-remove-nth-node-from-end-of-list.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
+---
+id: remove-nth-node-from-end-of-list
+title: Remove nth node from end of list (LeetCode)
+sidebar_label: 0019-remove-nth-node-from-end-of-list
+tags:
+  - Two Pointers
+  - Linked List
+description: Given the head of a linked list, remove the nth node from the end of the list and return its head.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0000-0099/0019-remove-nth-node-from-end-of-list.md:8]: replace 

`description: Given the head of a linked list, remove the nth node from the end of the list and return its head.`

to

`description: "Given the head of a linked list, remove the nth node from the end of the list and return its head."`

File: dsa-solutions/lc-solutions/0000-0099/0019-remove-nth-node-from-end-of-list.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+- **Input:** `head = [1], n = 1`
+- **Output:** `[]`
+
+#### Example 3
+
+- **Input:** `head = [1,2], n = 1`
+- **Output:** `[1]`
+
+### Constraints
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0000-0099/0019-remove-nth-node-from-end-of-list.md:38]: replace 

```md
### Constraints

- The number of nodes in the list is sz.
- `1 <= sz <= 30`
- `0 <= Node.val <= 100`
- `1 <= n <= sz`
```

TO

```md
### Constraints

- The number of nodes in the list is sz.
- $1 <= sz <= 30$
- $0 <= Node.val <= 100$
- $1 <= n <= sz$
```

File: dsa-solutions/lc-solutions/0000-0099/0020-valid-parentheses.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
+---
+id: valid-parentheses
+title: Valid Parentheses (LeetCode)
+sidebar_label: 0020-valid-parentheses
+tags:
+  - Stack
+
+description: Given a string s containing just the characters '(', ')', '{', '}', '[' and ']', determine if the input string is valid.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0000-0099/0020-valid-parentheses.md:8]: REPLACE

```md
description: Given a string s containing just the characters '(', ')', '{', '}', '[' and ']', determine if the input string is valid.
```

TO

```md
description: "Given a string s containing just the characters '(', ')', '{', '}', '[' and ']', determine if the input string is valid."
```

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

**Comment by Hemav009** [comment]
@Ajay-Dhangar   Updated

**Review by Ajay-Dhangar — APPROVED** [review]
done

**Pull request merged**